### DISCUSSION TITLE
function "context" not defined

### DISCUSSION THREAD

**Discussion by howdu** [hugo-mods/icons#10] [Q&A]
`{{ partial "icon" (context . name "fas heart") }}`

This semi-works 
`{{ partial "icon" (dict "context" . "name" "fas heart") }}`

But svg is not included even though it's last. 
> mod-icons: no used icons found. please ensure that the 'icon-data' partial is loaded LAST in your body!

Checking the html is added inline correctly but not the referenced svg:
`<svg class="icon"><use xlink:href="#icon-heart"></use></svg>`

**Comment by kdevo** [comment]
```
{{ partial "icon" (context . name "fas heart") }}
```

I'm afraid that this is wrongly documented in the README. If you want you can fix it via PR.
Otherwise I'll keep track of it via this issue.

> But svg is not included even though it's last.

Did you include the `icon-data` shortcode as the warning suggests? Like the following:

```
{{ partial "icon-data" . }}
```

If still not working, could you share a minimal working example for reproducing this?

**Comment by howdu** [comment]
Fix the readme but I've realised it's not working because I'm using partial includes to build up my pages. Is there a way of making it work in that way? 

https://github.com/hugo-mods/icons/pull/8

**Comment by kdevo** [comment]
Hi @howdu,

Thanks for the PR.
Do you have any possibility of passing the context around in these partials from one to another? Sometimes it's enough to just call your partial via `{{ partial "example" . }}`. 

Why is this required? The context is used to pass a "bag" with all the icons that are used on site. This is needed because the SVG data is rendered all at once, at a different place by default.

For complicated setups I would recommend the [Data Pattern](https://github.com/hugo-mods/icons#data-pattern) which requires maintaining a separate `icons.yaml` file but is more robust in most cases.

Last resort alternative would be to use the [inline loader](https://github.com/hugo-mods/icons/blob/main/layouts/partials/loaders/svg-inline.html), but then you'll loose some advantages of the symbol approach (`svg use`).

Hope this gives some insights.
